src/main/java/org/saltyrtc/tasks/webrtc/messages/Candidate.java

```diff
@@ -10,33 +10,79 @@
 
 import org.saltyrtc.client.annotations.NonNull;
 import org.saltyrtc.client.annotations.Nullable;
+import org.saltyrtc.client.exceptions.ValidationError;
+import org.saltyrtc.client.helpers.ValidationHelper;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
 
 /**
  * Immutable POJO containing candidate information.
  */
-class Candidate {
-    @NonNull
-    private final String candidate;
-    @Nullable
-    private final String sdpMid;
-    @Nullable
-    private final Integer sdpMLineIndex;
-
-    public Candidate(@NonNull String candidate, @Nullable String sdpMid, @Nullable Integer sdpMLineIndex) {
+public class Candidate {
+    @NonNull private static final String FIELD_CANDIDATE = "candidate";
+    @NonNull private static final String FIELD_SDP_MID = "sdpMid";
+    @NonNull private static final String FIELD_SDP_M_LINE_INDEX = "sdpMLineIndex";
+
+    @NonNull private final String candidate;
+    @Nullable private final String sdpMid;
+    @Nullable private final Integer sdpMLineIndex;
```

Comment:
This is nullable (as specified by the protocol), but the `sdpMLineIndex` parameter of the `IceCandidate` constructor wants an `int`. Any suggestion what to do if a candidate without an `sdpMLineIndex` would arrive? Throw away the candidate, or can this value be faked somehow? (I don't really know what `sdpMLineIndex` does.)

Reply:
The index refers to the media line. IIRC it's only used if you don't provide a `mid` which is preferred. So, it looks up the corresponding media line by the `mid` and falls back to the index of the media line (within the remote description's SDP) if not provided.

Reply:
For data channels, this is always `0` unless someone tries to negotiate audio/video with the web client...